icaruscode/PMT/OpReco/ICARUSFlashAssAna_module.cc

```diff
@@ -0,0 +1,457 @@
+
+
+  // We work out the trigger information here 
+  if( !fTriggerLabel.empty() ) { 
+
+      art::Handle<std::vector<sim::BeamGateInfo>> beamgate_handle;
+      e.getByLabel( fTriggerLabel, beamgate_handle );
+
+      if( beamgate_handle.isValid() ) {
+
+        for( auto const & beamgate : *beamgate_handle ) {
+
+          m_beam_gate_start = beamgate.Start(); 
+          m_beam_gate_width = beamgate.Width(); 
+          m_beam_type = beamgate.BeamType() ;
+
+        }
+
+      }
+
+      else {
+        std::cout << "Invalid Trigger Data product " << fTriggerLabel.label() << "\n" ;
+      }
+
+  }
+
+  else {
+     std::cout << "Trigger Data product " << fTriggerLabel.label() << " not found!\n" ;
+  }
+
+
+  // Now we take care of the labels
+  if ( !fFlashLabels.empty() ) {
+
+    for ( size_t i=0; i<fFlashLabels.size(); i++  ) {
+
+      auto const label = fFlashLabels[i];
+
+      art::Handle<std::vector<recob::OpFlash>> flash_handle;
+      e.getByLabel( label, flash_handle );
+
+      if( flash_handle.isValid() ) {
+
+        art::FindManyP<recob::OpHit> ophitsPtr( flash_handle, e, label );
+
+        for ( size_t idx=0; idx<flash_handle->size(); idx++ ) {
+
+          m_flash_id = idx;
+          auto const & flash = (*flash_handle)[idx];
+
+          m_flash_time = flash.Time();
+          m_sum_pe = flash.TotalPE();
+            
+          auto const & ophits = ophitsPtr.at(idx);
+
+          // Get the multiplicity, the position and the number of PE per Side
+          float xyz[3] = {0.0, 0.0, 0.0};
+          processOpHits( ophits, 
+                         m_multiplicity_left, m_multiplicity_right, 
+                         m_sum_pe_left, m_sum_pe_right, xyz, fOpHitTrees[i] );
+
+          /*
+          std::cout << "\tflash id: " << idx << ", time: " << m_flash_time;
+          std::cout << ", multiplicity left: " << m_multiplicity_left << ", multiplicity right: " << m_multiplicity_right;
+          std::cout << ", sum pe left: " << m_sum_pe_left << ", sum pe right: " << m_sum_pe_right;
+          std::cout << " coor: [" << xyz[0] << ", " << xyz[1] << ", " << xyz[2] << "]";
+          std::cout << " coor: [" << 0.0 << ", " << flash.YCenter() << ", " << flash.ZCenter() << "]";
+          std::cout << " coor: [" << 0.0 << ", " << flash.YWidth() << ", " << flash.ZWidth() << "]";
+	        std::cout  << "\n";
+          */
+
+          m_multiplicity = m_multiplicity_left+m_multiplicity_right;
+
+          //m_flash_x = 0.0;
+          //m_flash_width_x = 0.0;
+          m_flash_y = flash.YCenter(); 
+          m_flash_width_y = flash.YWidth();
+          m_flash_z = flash.ZCenter();
+          m_flash_width_z = flash.ZWidth();
+
+          fOpFlashTrees[i]->Fill();
+
+        }
+
+      }
+
+      else {
+
+        mf::LogError("ICARUSFlashAssAna")
+           << "Invalid recob::OpFlash with label"+label.label()+"\n";
+      }
+
+    } // end for on flash input tags
+
+  }
+
+  else {
+    mf::LogError("ICARUSFlashAssAna")
```

Comment:
It's weird that a mis-configuration would log an error for every single event -- maybe move this to the constructor?